cmd/go-fix-acronym/rewrite.go

```diff
@@ -37,10 +37,34 @@ func rewriteNode(pattern *regexp.Regexp, node ast.Node) {
 			if !ok {
 				return true
 			}
-			name := pattern.ReplaceAllFunc([]byte(ident.Name), func(b []byte) []byte {
-				return []byte(strings.ToUpper(string(b)))
-			})
-			ident.Name = string(name)
+
+			var result string
+
+			matches := pattern.FindAllStringSubmatchIndex(ident.Name, -1)
+			if matches == nil {
+				return true
+			}
+
+			var match []int
+			for _, match = range matches {
+				// No submatches.
+				if len(match) == 2 {
+					result += ident.Name[0:match[0]]
```

Comment:
This doesn't behave right when there are multiple matches. I had to try for myself

```.diff
diff --git a/cmd/go-fix-acronym/rewrite_test.go b/cmd/go-fix-acronym/rewrite_test.go
index 7bc65ab..2966b1f 100644
--- a/cmd/go-fix-acronym/rewrite_test.go
+++ b/cmd/go-fix-acronym/rewrite_test.go
@@ -59,6 +59,12 @@ func TestRewriteSimple(t *testing.T) {
                        expected: "//hello\npackage main\n\nfunc GetCPU() {}",
                        c:        config{acronyms: []string{"Cpu"}},
                },
+               {
+                       name:     "simple",
+                       input:    "//hello\npackage main\nfunc GetCpuFromCpuList(){}",
+                       expected: "//hello\npackage main\n\nfunc GetCPUFromCPUList() {}",
+                       c:        config{acronyms: []string{"Cpu"}},
+               },
                {
                        name: "group",
                        input: `package main
```

```
$ go test ./cmd/go-fix-acronym
--- FAIL: TestRewriteSimple (0.00s)
    --- FAIL: TestRewriteSimple/simple#01 (0.00s)
        rewrite_test.go:45: expected "//hello\npackage main\n\nfunc GetCPUFromCPUList() {}\n", but got "//hello\npackage main\n\nfunc GetCPUGetCpuFromCPUList() {}\n"
FAIL
FAIL	github.com/containerd/protobuild/cmd/go-fix-acronym	0.004s
FAIL
```

Reply:
Thanks. It should be fixed in the last revision.